### DISCUSSION TITLE
Injeção de dependência fora do controller

### DISCUSSION THREAD

**Discussion by rodolfomuller** [balta-io/2811#24] [Q&A] [answered]
Olá!
Tenho uma dúvida com a injeção de dependência quando a chamada da classe que tem a injeção é chamada de outra classe, pois só vejo exemplos usando a partir do controller, somente é possível fazer por ele?
Tenho que criar as variáveis para receber os valores e criar um construtor no controller e ir passando de construtor para construtor até chegar na classe que vou utilizar realmente?

**Comment by andrebaltieri** [comment, marked as answer]
Boa noite, @rodolfomuller, como vai?

Um dos lugares onde a DI é mais utilizada é sim nos controladores, porém você pode utilizar ela em qualquer classe que dê suporte a resolução de dependências do ASP.NET. Se você tiver um exemplo de onde quer resolver estas dependências, posso te direcionar melhor.

Sobre os construtores, sim, esta é a forma que temos de fazer. Em adicional, nos controladores, podemos utilizar o `[FromServices]` para casos mais pontuais.

**Comment by rodolfomuller** [comment]
Olá Balta!
Obrigado pelo retorno. 
Eu fiz um exemplo, mas acho que está faltando alguma coisa no projeto, pois não está rodando.
Vamos ver se eu consigo demonstrar, tenho um controller e 3 classes, uma chama a outra, até chegar na classe3 onde estou aplicando a DI, só que eu trago desde o controller passando por parâmetro para classe1, classe2 e classe3 a varíavel da classe3, não sei se é o correto e me parece bem custoso fazer desta forma.

Segue abaixo o exemplo que tentei fazer:
https://github.com/rodolfomuller/TesteDI
